### PR TITLE
[wasm] Fix AOT support.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -5892,7 +5892,7 @@ ves_icall_Mono_Runtime_DumpStateSingle (guint64 *portable_hash, guint64 *unporta
 
 
 ICALL_EXPORT void
-ves_icall_Mono_Runtime_RegisterReportingForNativeLib (const char *path_suffix, const char *module_name)
+ves_icall_Mono_Runtime_RegisterReportingForNativeLib (const char *path_suffix, const char *module_name, MonoError *error)
```
Comment (jaykrell) [inline mono/metadata/icall.c]: Alternatively, mark it nohandles().

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins Windows x64 C++

**Comment by vargaz** [comment]
@monojenkins build Windows x64 C++

**Comment by vargaz** [comment]
@monojenkins build failed

File: sdks/builds/wasm.mk [hunk 2]
```diff
@@ -103,6 +103,8 @@ _wasm-$(1)_CONFIGURE_FLAGS= \
 	--enable-maintainer-mode \
 	--enable-minimal=appdomains,com,remoting \
 	--enable-icall-symbol-map \
+	--with-cooperative-gc=no \
+	--enable-hybrid-suspend=no \
```
Comment (lewurm) [inline sdks/builds/wasm.mk]: is there a problem around safepoints?

Comment (vargaz) [reply sdks/builds/wasm.mk]: the device runtime is compiled to not use them, so its missing the relevant icall so it asserts when loading aot-ed code.

File: sdks/wasm/library_mono.js [hunk 3]
```diff
@@ -80,7 +80,19 @@ var MonoSupportLib = {
 		},
 
 		mono_load_runtime_and_bcl: function (vfs_prefix, deploy_prefix, enable_debugging, file_list, loaded_cb, fetch_file_cb) {
-			Module.FS_createPath ("/", vfs_prefix, true, true);
+			// /dev/random doesn't work on js shells, so define our own
+			// See library_fs.js:createDefaultDevices ()
+			var random_device;
+			if (typeof crypto !== 'undefined') {
+				var randomBuffer = new Uint8Array(1);
```
Comment (kumpera) [inline sdks/wasm/library_mono.js]: This is plain bad code.
We're littering `mono_load_runtime_and_bcl` with something completely unrelated.

mono-random.c should have a `HOST_WASM` version that does invoke into this code, which should be a function by itself.

**Pull request merged**